samza/sso-account-updater/src/main/java/org/sunbird/jobs/samza/service/SSOAccountUpdaterService.java

```diff
@@ -0,0 +1,183 @@
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.jobs.samza.util.JobLogger;
+import org.sunbird.jobs.samza.util.JSONUtils;
+import org.sunbird.jobs.samza.util.SSOAccountUpdaterParams;
+
+import okhttp3.OkHttpClient;
+import okhttp3.MediaType;
+import okhttp3.RequestBody;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import javax.ws.rs.core.HttpHeaders;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.List;
+
+public class SSOAccountUpdaterService {
+
+    private static MediaType jsonMediaType = MediaType.parse(javax.ws.rs.core.MediaType.APPLICATION_JSON);
+    private JobLogger Logger = new JobLogger(SSOAccountUpdaterService.class);
+    private Config appConfig = null;
+    private OkHttpClient client = new OkHttpClient();
+    private Gson gson = new Gson();
+
+    public void initialize(Config config) throws Exception {
+        JSONUtils.loadProperties(config);
+        appConfig = config;
+        Logger.info("SSOAccountUpdaterService:initialize: Service config initialized");
+    }
+
+    public void processMessage(Map<String, Object> message) throws Exception {
+
+        Map<String, Object> eventMap = (Map<String, Object>) message.get(SSOAccountUpdaterParams.event.name());
+
+        String userExternalId = (String) eventMap.get(SSOAccountUpdaterParams.userExternalId.name());
+        String nameFromPayload = (String) eventMap.get(SSOAccountUpdaterParams.nameFromPayload.name());
+        String channel = (String) eventMap.get(SSOAccountUpdaterParams.channel.name());
+        String orgExternalId = (String) eventMap.get(SSOAccountUpdaterParams.orgExternalId.name());
+        List<String> passedRoles = (List<String>) eventMap.get(SSOAccountUpdaterParams.roles.name());
+        String userId = (String) eventMap.get(SSOAccountUpdaterParams.userId.name());
+        List<Map<String, Object>> organisations = (List<Map<String, Object>>) eventMap.get(SSOAccountUpdaterParams.organisations.name());
+        String firstName = (String) eventMap.get(SSOAccountUpdaterParams.firstName.name());
+
+        Logger.info("SSOAccountUpdaterService:processMessage: Processing user data for userId - " + userId);
+
+        Map<String, Object> passedOrg = getOrg(channel, orgExternalId);
+        if (MapUtils.isEmpty(passedOrg)) {
+            ProjectCommonException.throwClientErrorException(ResponseCode.invalidOrgData);
+        }
+
+        Map<String, Object> updateMap = new HashMap<>();
+        if (!StringUtils.equals(nameFromPayload, firstName)) {
+            updateMap.put(SSOAccountUpdaterParams.firstName.name(), nameFromPayload);
+        }
+
+        if (CollectionUtils.isEmpty(passedRoles)) {
+            passedRoles = new ArrayList<String>();
+            passedRoles.add(SSOAccountUpdaterParams.PUBLIC.name());
+        }
+
+        String passedOrgId = (String) passedOrg.get(SSOAccountUpdaterParams.identifier.name());
```

Comment:
This method is too long. Can be broken into helper methods for code maintainability.

Reply:
Review comment addressed in code